platforms/software/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/JavaApplicationInitIntegrationTest.groovy

```diff
@@ -274,4 +274,21 @@ class JavaApplicationInitIntegrationTest extends AbstractJvmLibraryInitIntegrati
         expect:
         succeeds('init', '--type', 'java-application', '--dsl', 'groovy')
     }
+
+    def "creates with gradle/libs.versions.toml when using #scriptDsl build scripts"() {
+        when:
+        run ('init', '--type', 'java-application', '--dsl', scriptDsl.id)
+
+        then:
+        versionsCatalogGenerated()
```

Comment:
Yes, the intent is to use version catalogs always when there are dependency declarations to encourage version catalog usage since it became a stable feature in Gradle 8.0.

If you generate a basic build it should not have a version catalog.

Reply:
I think it's probably fair to call a build with a few common dependencies (Apache utils, Guava) still pretty basic.  My concern is that this impacts Comprehensibility by making the Gradle learning curve steeper for newcomers.  They now have to understand the VC right from the beginning instead of discovering it as their build grows more complex - and at this point it's really a solution in search of a problem.  

I assume this approach has already been discussed and this decision made.  The larger number of things you need to understand just to get started and the lack of at least an opt-out for people already using `init` with the expectation that it doesn't generate a VC still bothers me though.

Reply:
I meant the `basic` template.

Reply:
Well since builds in `buildSrc` and `build-logic` don't automatically inherit the version catalog of the main build, the builds in there generate references to `../gradle/libs.versions.toml` and thus the version catalog is generated in all cases.

I should probably add the test for the version catalog existence in more places to make this fact more explicit.

Reply:
moved the existence check for the version catalog up to `ScriptDslFixture`, so this is done too